.github/workflows/ci.yml

```diff
@@ -0,0 +1,442 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: CI
```

Comment:
Is this identical to `build_wheels.yml`? What needs to be reviewed? why the name change?

Reply:
For easier review I made separate steps for changes and renaming. In the first commit you will see added jobs.
I renamed workflow filename and title since it will be more general CI workflow, building wheels are part of it of course, but keeping old name IMHO would be slightly misleading.